src/plugins/conditionWidget/components/ConditionWidget.vue

```diff
@@ -39,28 +39,95 @@ export default {
     inject: ['openmct', 'domainObject'],
     data: function () {
         return {
-            internalDomainObject: this.domainObject
+            conditionalLabel: '',
+            conditionSetIdentifier: null,
+            domainObjectLabel: '',
+            url: null,
+            urlDefined: false,
+            useConditionSetOutputAsLabel: false
         };
     },
     computed: {
-        urlDefined() {
-            return this.internalDomainObject.url && this.internalDomainObject.url.length > 0;
-        },
-        url() {
-            return this.urlDefined ? sanitizeUrl(this.internalDomainObject.url) : null;
+        label() {
+            return this.useConditionSetOutputAsLabel
+                ? this.conditionalLabel
+                : this.domainObjectLabel
+            ;
+        }
+    },
+    watch: {
+        conditionSetIdentifier: {
+            immediate: false,
+            handler: 'listenToConditionSetChanges'
         }
     },
     mounted() {
-        this.unlisten = this.openmct.objects.observe(this.internalDomainObject, '*', this.updateInternalDomainObject);
+        this.unlisten = this.openmct.objects.observe(this.domainObject, '*', this.updateDomainObject);
+        if (this.domainObject) {
+            this.updateDomainObject(this.domainObject);
+        }
     },
     beforeDestroy() {
         if (this.unlisten) {
             this.unlisten();
         }
+
+        this.stopListeningToConditionSetChanges();
     },
     methods: {
-        updateInternalDomainObject(domainObject) {
-            this.internalDomainObject = domainObject;
+        listenToConditionSetChanges() {
+            const self = this;
+            self.stopListeningToConditionSetChanges();
+
+            if (!self.conditionSetIdentifier) {
+                return;
+            }
+
+            self.openmct.objects.get(self.conditionSetIdentifier)
```

Comment:
I think we can safely use `await` here? might simplify the code a little. Also, I believe Vue component methods already have `this` bound to the component itself?